packages/agreement-process/src/services/agreementService.ts

```diff
@@ -39,21 +40,30 @@ import {
 
 const fileManager = initFileManager(config);
 
-const repository = eventRepository(
-  initDB({
-    username: config.eventStoreDbUsername,
-    password: config.eventStoreDbPassword,
-    host: config.eventStoreDbHost,
-    port: config.eventStoreDbPort,
-    database: config.eventStoreDbName,
-    schema: config.eventStoreDbSchema,
-    useSSL: config.eventStoreDbUseSSL,
-  }),
-  agreementEventToBinaryData
-);
-
-export const agreementService = {
-  async getAgreements(
+export class AgreementService {
+  private readModelService: ReadModelService;
+  private repository;
+
+  constructor(
+    readModelService: ReadModelService,
+    config: AgreementProcessConfig
+  ) {
+    this.readModelService = readModelService;
+    this.repository = eventRepository(
```

Comment:
same for dependency injection and class removal

Reply:
Updated